ADD: Project: AD353XR (#1544)
analogdevicesinc/hdl: docs/projects/ad353xr/index.rst

resulting in a full-scale output of 2.5 V (gain = 1) or 5 V (gain = 2) for a
reference voltage of 2.5 V. The :adi:`AD3530R` has an on-chip, buffered, 2.5 V
reference available at the VREF pin, capable of sourcing external loads up to
+5 mA.

Each DAC channel has its own Input register and DAC register. The DAC Register 
stores digital code equivalent to the DAC output voltage while the Input Register
acts as a temporary staging register before being passed on the DAC Register. With
the LDAC function, one or more DAC registers could be updated in parallel with
the data held in the Input Register. The DAC registers can also be directly written
to, in which the corresponding output updates immediately without an LDAC. 

The :adi:`AD3530R`/ :adi:`AD3530` contains eight buffered voltage output DAC 
channels capable of sourcing and sinking up to 40 mA of current.

The :adi:`AD3530R`/ :adi:`AD3530` contains a 27:1 multiplexer which could
output a voltage on the MUX_OUT pin that is a representative of
either the output voltage or output current of a chosen channel, or
the internal die temperature of the device.

> 

Reviewer: Remove one empty line

Author: removed

Applications:

* Optical transceivers
* Test and measurement
* Industrial automation

> * Data acquisition systems

Reviewer: Prefer - over * when creating a list

Author: changed

Supported boards
-------------------------------------------------------------------------------

> - :adi:`EVAL-AD3530RARDZ`

Reviewer: ```suggestion
- :adi:`EVAL-AD3530R`
```

Author: shortened

Supported devices
-------------------------------------------------------------------------------

- :adi:`AD3530`
- :adi:`AD3530R`

Supported carriers
-------------------------------------------------------------------------------

> - :xilinx:`ZedBoard <products/boards-and-kits/1-8dyf-11.html>` on FMCs
> - :xilinx:`Cora Z7-07S <products/boards-and-kits/1-1qlaz7n.html>` on GPIOs
> - :intel:`DE10-Nano <content/www/us/en/partner/showcase/offering/a5b3b0000004cbwAAA/de10nano-kit.html>` on GPIOs

Reviewer: + specify the PMOD for Cora and DE10-Nano

```suggestion
- :xilinx:`ZedBoard <products/boards-and-kits/1-8dyf-11.html>` on FMC
- :xilinx:`Cora Z7-07S <products/boards-and-kits/1-1qlaz7n.html>` on GPIOs
- :intel:`DE10-Nano <content/www/us/en/partner/showcase/offering/a5b3b0000004cbwAAA/de10nano-kit.html>` on GPIOs
```

Reviewer: Ohh, sorry, I thought it was on PMOD... Then you can leave things as they are now.

Block design
-------------------------------------------------------------------------------

Block diagram
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The data path is depicted in the below diagram:

.. image:: ad353xr_block_diagram.svg
   :width: 800
   :align: center
   :alt: AD353XR block diagram

Hardware setup
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

============ ================= ============== ================ ==============

> Signal       AD353XR Testpoint  ZedBoard FMC  Cora Z7-07S GPIO DE10-Nano GPIO

Reviewer: ```suggestion
Signal       AD353XR Testpoint ZedBoard FMC   Cora Z7-07S GPIO  DE10-Nano GPIO
```

Author: updated

============ ================= ============== ================ ==============
CSB(SS0)     PMOD P1           M19/FMC-LA00_P F16              AE19
SDO(MOSI)    PMOD P2           N19/FMC-LA01_P T12              AG15
SDI(MISO)    PMOD P3           N20/FMC-LA01_N W15              AF18
SCK          PMOD P4           D18/FMC-CLK1_P H15              AG18
RESETB       PMOD P8           T19/FMC-LA10_N V13              AE20
LDACB        PMOD P9           J18/FMC_LA05_P T14              AE17
============ ================= ============== ================ ==============

> The evaluation board is powered by 5 V voltage from an external USB.

Reviewer: ```suggestion
.. important::

   The evaluation board is powered by 5V voltage from an external USB.
```

Author: updated

Reviewer: Forgot to highlight this, but the text from inside the "important" role, should be indented with 3 spaces. I thought this would be highlighted when I suggested the change

Author: forgot to double check too but this is updated now.

Hardware related (Links To be Updated)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

- Product datasheet: :adi:`AD353XR`
- `UG-1203: EVAL-AD3530RARDZ Board User Guide <https://www.analog.com/media/en/technical-documentation/user-guides/EVAL-AD4110-1SDZ-UG-1203.pdf>`__

HDL related
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

- :git-hdl:`AD353XR HDL project source code <projects/ad353xr>`

.. list-table::
   :widths: 30 35 35
   :header-rows: 1

   * - IP name
     - Source code link
     - Documentation link
   * - AXI_CLKGEN
     - :git-hdl:`library/axi_clkgen`
     - :ref:`axi_clkgen`
   * - AXI_DMAC
     - :git-hdl:`library/axi_dmac`
     - :ref:`axi_dmac`
   * - AXI_HDMI_TX
     - :git-hdl:`library/axi_hdmi_tx`
     - :ref:`axi_hdmi_tx`
   * - AXI_I2S_ADI
     - :git-hdl:`library/axi_i2s_adi`
     - ---
   * - AXI_SPDIF_TX
     - :git-hdl:`library/axi_spdif_tx`
     - ---
   * - AXI_SYSID
     - :git-hdl:`library/axi_sysid`
     - :ref:`axi_sysid`
   * - AXI_SYSID_ROM
     - :git-hdl:`library/sysid_rom`
     - :ref:`axi_sysid`
   * - UTIL_I2C_MIXER
     - :git-hdl:`library/util_i2c_mixer`
     - ---

> Software related (Links To be Updated)

Reviewer: ```suggestion
Software related
```

Author: removed

- All supported boards uses the same HDL GPIO EMIO Number

.. list-table::
   :widths: 25 25 25 25
   :header-rows: 2

   * - GPIO signal
     - Direction
     - HDL GPIO EMIO
     - Software GPIO
   * -
     - (from FPGA view)
     -
     - Zynq-7000
   * - RESETB
     - OUT
     - 33
     - 87
   * - LDACB
     - OUT
     - 34
     - 88

Building the HDL project
-------------------------------------------------------------------------------

The design is built upon ADI's generic HDL reference design framework.
ADI distributes the bit/elf files of these projects as part of the
:dokuwiki:`ADI Kuiper Linux <resources/tools-software/linux-software/kuiper-linux>`.
If you want to build the sources, ADI makes them available on the
:git-hdl:`HDL repository </>`. To get the source you must
`clone <https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository>`__
the HDL repository, and then build the project as follows:

**Linux/Cygwin/WSL**

.. shell::

   $cd hdl/projects/ad353xr/zed
   $make

A more comprehensive build guide can be found in the :ref:`build_hdl` user guide.

Resources
-------------------------------------------------------------------------------

> Hardware related (Links To be Updated)
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 
> - Product datasheet: :adi:`AD353XR`

Reviewer: ```suggestion
Hardware related
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

- Product datasheet: :adi:`AD3530R`
```
Will there be a chip called AD353**X**R? Or should it be 353**0**R?

Author: This is a family of chips but ad3530r will be releasing first as per the requestor/apps engg
It was decided to name this as ad353xr despite for now having only 1 member chip
